src/views/UserSearch/UserSearch.js

```diff
@@ -106,13 +110,32 @@ class UserSearch extends React.Component {
     }));
   }
 
+  generateOverdueLoanReport = props => {
+    const {
+      reset,
+      GET,
+    } = props.mutator.loans;
+    const { exportInProgress } = this.state;
+
+    if (exportInProgress) {
+      return;
+    }
+
+    this.setState({ exportInProgress: true }, () => {
+      reset();
+      GET()
+        .then(loans => this.overdueLoanReport.toCSV(loans))
+        .then(this.setState({ exportInProgress: false }));
```

Comment:
```suggestion
        .then(() => this.setState({ exportInProgress: false }));
```
Please change it the way I suggested. I don't know how it turned out to work just that way.

Reply:
🤦‍♂ Good catch. Thank you!